tests/api/test_agents.py

```diff
@@ -0,0 +1,108 @@
+import pendulum
+import pytest
+
+from prefect import api, models
+from prefect_server import config
+
+
+class TestCreateAgent:
+    async def test_register_agent(self, tenant_id):
+        # create an agent
+        agent_id = await api.agents.register_agent(
+            tenant_id=tenant_id,
+            labels=["foo", "bar"],
+        )
+        agent = await models.Agent.where(id=agent_id).first(
+            {"id", "tenant_id", "labels"}
+        )
+        assert agent.id == agent_id
+        assert agent.tenant_id == tenant_id
+        assert set(agent.labels) == set(["bar", "foo"])
+
+    @pytest.mark.parametrize("labels", [[], None])
+    async def test_register_agent_with_empty_labels(self, tenant_id, labels):
+        # create an agent
+        agent_id = await api.agents.register_agent(
+            tenant_id=tenant_id,
+            labels=labels,
+        )
+        agent = await models.Agent.where(id=agent_id).first(
+            {"id", "tenant_id", "labels"}
+        )
+        assert agent.id == agent_id
+        assert agent.tenant_id == tenant_id
+        assert agent.labels == []
+
+    async def test_register_agent_with_optional_arguments(self, tenant_id):
+        labels = ["foo", "bar"]
+        agent_id = await api.agents.register_agent(
+            tenant_id=tenant_id, name="MyNewAgent", type="NewAgent", labels=labels
+        )
+        agent = await models.Agent.where(id=agent_id).first(
+            {"id", "tenant_id", "labels", "name", "type"}
+        )
+        assert agent.id == agent_id
+        assert agent.tenant_id == tenant_id
+        assert agent.name == "MyNewAgent"
+        assert agent.type == "NewAgent"
+        assert set(agent.labels) == set(labels)
+
+    async def test_register_multiple_agents(self, tenant_id):
```

Comment:
Could you add another test similar to this one where you register an agent with the same 3 labels in different orders and confirm that only 1 new agent is actually created in the DB + the same ID is returned each time?